### PR TITLE
TFKUBE-109: default tags

### PR DESCRIPTION
**Pull request by jjeongatl**
Default tags added on aws providers
remove all resource tags but we can still add any extra tags on leaf module level

### CODE REVIEW

**Comment by jjeongatl** [comment]
I created asg_ec2_tagging module for workaround. The module will be initialised only when infra is ready. It uses output of the terraform result to fetch ASG and EC2 information.

File: pkg/scripts/uninstall.sh [hunk 1]
```diff
@@ -151,6 +151,15 @@ destroy_tfstate() {
...
 
+cleanup_asg_ec2_tagging_module() {
+  echo "Cleaning up asg_ec2_tagging Module."
+  cd "${SCRIPT_PATH}/../modules/AWS/asg_ec2_tagging"
+  rm -rf *.tfvars
+  rm -rf .terraform
+  rm -rf .terraform.lock.hcl
+  rm -rf terraform.tfstate
+  rm -rf terraform.tfstate.backkup
+}
```
Comment (nghazali) [inline pkg/scripts/uninstall.sh:162]: Move this stuff into `cleanup.sh`

File: pkg/scripts/uninstall.sh [hunk 2]
```diff
@@ -151,6 +151,15 @@ destroy_tfstate() {
   fi
 }
 
+cleanup_asg_ec2_tagging_module() {
+  echo "Cleaning up asg_ec2_tagging Module."
+  cd "${SCRIPT_PATH}/../modules/AWS/asg_ec2_tagging"
```
Comment (nghazali) [inline pkg/scripts/uninstall.sh:156]: If you change the current folder you need to make sure you back to the same folder when you done with cleaning. I mean your script should not change the current folder before the script.

File: pkg/scripts/install.sh [hunk 3]
```diff
@@ -171,6 +171,15 @@ set_current_context_k8s() {
   echo
 }
 
+add_tags_to_asg_resources() {
+  echo "Tagging Auto Scailing Group and EC2 instances."
+  CHDIR="-chdir=${SCRIPT_PATH}/../modules/AWS/asg_ec2_tagging"
+  cp "${SCRIPT_PATH}/../../${CONFIG_FILE}" "${SCRIPT_PATH}/../modules/AWS/asg_ec2_tagging"
+
+  terraform "${CHDIR}" init
+  terraform "${CHDIR}" apply -auto-approve "-var-file=${CONFIG_FILE}" -var s3_bucket="${S3_BUCKET}" -var bucket_key="${BUCKET_KEY}/terraform.tfstate"  -compact-warnings
```
Comment (nghazali) [inline pkg/scripts/install.sh:180]:
```suggestion
  ASG_EC2_TAG_DIR="${SCRIPT_PATH}/../modules/AWS/asg_ec2_tagging"
  cp "${SCRIPT_PATH}/../../${CONFIG_FILE}" "${ASG_EC2_TAG_DIR}"

  terraform -chdir="${ASG_EC2_TAG_DIR}" init
  terraform -chdir="${ASG_EC2_TAG_DIR}" apply -auto-approve "${OVERRIDE_CONFIG_FILE}" -var s3_bucket="${S3_BUCKET}" -var bucket_key="${BUCKET_KEY}/terraform.tfstate"  -compact-warnings
```

**Review by nghazali — CHANGES_REQUESTED** [review]

**Review by nghazali — APPROVED** [review]

**Review by nanux — APPROVED** [review]

**Pull request merged**